test/NuGet.Core.FuncTests/NuGet.Commands.FuncTest/RestoreCommandTests.cs

```diff
@@ -2023,11 +2027,13 @@ public async Task RestoreCommand_RestoreFloatingVersionWithFailingLocalSourceAsy
                     LockFilePath = Path.Combine(projectDir, "project.lock.json")
                 };
 
+                // Act & Assert
                 var command = new RestoreCommand(request);
+                RestoreResult result = await command.ExecuteAsync();
+                await result.CommitAsync(logger, CancellationToken.None);
 
-                // Act & Assert
-                var ex = await Assert.ThrowsAsync<FatalProtocolException>(async () => await command.ExecuteAsync());
-                Assert.NotNull(ex);
+                Assert.False(result.Success);
+                Assert.Contains("NU1302", string.Join(Environment.NewLine, logger.ErrorMessages));
```

Comment:
Why this NewLine?

Reply:
copied that pattern from somewhere nearby. likely easier to understand the serarate errors when the assert fails.